addOns/ascanrulesBeta/src/test/java/org/zaproxy/zap/extension/ascanrulesBeta/XxeScanRuleUnitTest.java

```diff
@@ -121,4 +129,225 @@ public void replaceMultipleElementsAndRemoveHeader() {
+        assertThat(countMessagesSent, equalTo(1));
+    }
+
+    @Test
+    public void shouldSendThreeMessagesForLocalFileInclusionRule() throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response =
+                                "<foo>root:*:0:0:System Administrator:/var/root:/bin/sh</foo>";
+                        return newFixedLengthResponse(
+                                NanoHTTPD.Response.Status.OK, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setMethod("POST");
+        msg.getRequestHeader().setHeader("Content-Type", "application/xml");
+        rule.init(msg, parent);
+
+        // When
+        rule.localFileInclusionAttack(msg);
+
+        // Then
+        assertThat(countMessagesSent, equalTo(3));
+    }
+
+    @ParameterizedTest
+    @EnumSource(
+            value = NanoHTTPD.Response.Status.class,
+            names = {"OK", "BAD_REQUEST"})
+    public void shouldAlertWhenLocalFileIncludedInResponse(NanoHTTPD.Response.Status status)
+            throws Exception {
+        // Given
+        String test = "/test";
+        this.nano.addHandler(
+                new NanoServerHandler(test) {
+                    @Override
+                    protected NanoHTTPD.Response serve(NanoHTTPD.IHTTPSession session) {
+                        consumeBody(session);
+                        String response = "root:*:0:0:System Administrator:/var/root:/bin/sh";
+                        return newFixedLengthResponse(status, NanoHTTPD.MIME_PLAINTEXT, response);
+                    }
+                });
+        HttpMessage msg = this.getHttpMessage(test);
+        msg.setRequestBody("<?xml version=\"1.0\"?><comment><text>test</text></comment>");
+        msg.getRequestHeader().setMethod("POST");
+        msg.getRequestHeader().setHeader("Content-Type", "application/xml");
+        rule.init(msg, parent);
+
+        // When
+        rule.localFileInclusionAttack(msg);
+
+        // Then
+        assertThat(alertsRaised.size(), equalTo(1));
+        assertThat(alertsRaised.get(0).getEvidence(), equalTo("root:*:0:0"));
+        assertThat(
+                alertsRaised.get(0).getAttack(),
+                equalTo(
+                        MessageFormat.format(XxeScanRule.ATTACK_HEADER, "file:///etc/passwd")
+                                + XxeScanRule.ATTACK_BODY));
+        assertThat(alertsRaised.get(0).getRisk(), equalTo(Alert.RISK_HIGH));
+        assertThat(alertsRaised.get(0).getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
```

Comment:
```suggestion
        Alert alert = alertsRaised.get(0);
        assertThat(alert.getEvidence(), equalTo("root:*:0:0"));
        assertThat(
                alert.getAttack(),
                equalTo(
                        MessageFormat.format(XxeScanRule.ATTACK_HEADER, "file:///etc/passwd")
                                + XxeScanRule.ATTACK_BODY));
        assertThat(alert.getRisk(), equalTo(Alert.RISK_HIGH));
        assertThat(alert.getConfidence(), equalTo(Alert.CONFIDENCE_MEDIUM));
```

This likely applies elsewhere as well.

Reply:
This has been updated as well.